dspace-jspui/src/main/java/org/dspace/app/webui/servlet/FeedServlet.java

```diff
@@ -267,17 +269,23 @@ else if ( ! itemsChanged(context, dso, cFeed.timeStamp))
     private boolean itemsChanged(Context context, DSpaceObject dso, long timeStamp)
             throws SQLException
     {
-        // construct start and end dates
+        /*
+         * Construct start and end dates.
+         *
+         * According to the Solr DateField documentation: "date field shall be 
+         * of the form 1995-12-31T23:59:59Z The trailing "Z" designates UTC time
+         * and is mandatory (See below for an explanation of UTC). Optional
+         * fractional seconds are allowed, as long as they do not end in a
+         * trailing 0 (but any precision beyond milliseconds will be ignored).
+         * All other parts are mandatory." Therefore the Full ISO 8601 format
+         * is e.g. "2009-07-16T13:59:21Z" is maintained
+         */
         DCDate dcStartDate = new DCDate( new Date(timeStamp) );
         DCDate dcEndDate = new DCDate( new Date(System.currentTimeMillis()) );
-
```

Comment:
@tdonohue I have in mind that you was juggling with this sometime ago. Right? Can you please take a look?

Reply:
No, I don't recall working on this recently.  That said, the code here looks reasonable enough to me..but likely needs testing, as there seem to be significant changes to the Solr queries (and I'll admit, I'm not familiar with these Solr queries myself).

Reply:
I finally found what I remembered: #1146 (https://jira.duraspace.org/browse/DS-2542). That PR handles timestamps in SOLR but is not related to this one.

Reply:
To ease a review: 
These changes corrects the syntaxes and requirements of Solr as we use it. If you take a look on how locations are indexed here: https://github.com/DSpace/DSpace/blob/716912f275b233522dcab982d049e0f01b9a0b60/dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java#L694 you will notice that a "m" or "l" is added to the ID of Community/Collection.
As for the lastModified field, Solr doesn't support comparison operators such as "<=, =>, ..."  for Dates. For more information, see: https://lucene.apache.org/solr/guide/6_6/working-with-dates.html

Reply:
Code deployed in production here. It fixes the error messages we had in dspace.log and RSS feeds are correctly displaying.